src/ethereum_spec_tools/evm_tools/utils.py

```diff
@@ -172,3 +172,15 @@ def secp256k1_sign(msg_hash: Hash32, secret_key: int) -> Tuple[U256, ...]:
         U256.from_be_bytes(signature[32:64]),
         U256(signature[64]),
     )
+
+
+def encode_to_hex(data: Union[bytes, int]) -> str:
+    """
+    Encode the data to a hex string.
+    """
+    if isinstance(data, int):
+        return hex(data)
+    elif isinstance(data, bytes):
+        return "0x" + data.hex()
+    else:
+        raise Exception("Invalid data type")
```

Comment:
Should this be in `src/ethereum/utils/hexadecimal.py`?

Reply:
I am not sure if this will be useful to have in the specs code directly since we do not encode anything to hex string. This is exclusively used in the `t8n` tool for producing outputs.

Reply:
Woops, you're entirely correct!